code/modules/vehicles/vehicle.dm

```diff
@@ -0,0 +1,403 @@
+					"<span class='danger'>You hit [src] with the [I.name]</span>")
+
+/obj/vehicle/proc/take_damage(amount, type="brute")
+	if(amount)
+		var/damage = absorbDamage(amount,type)
+		health -= damage
+		update_health()
+
+/obj/vehicle/proc/update_health()
+	if(src.health > 0)
+
+	else
+		qdel(src)
+
+/obj/vehicle/proc/compareParts(var/obj/item/vehicle_parts/input, var/list/part_list) //Meant for checking parts and whether there is already one installed or not.
+	for(var/obj/item/F in part_list)
+		var/obj/item/vehicle_parts/N = F
+		if(input.part_type == "seat") //Seats are ignored, of course.
+			return 0
+		else if(input.part_type == N.part_type)
+			return 1
+	return 0
+
+
+
+
+
+
+/obj/vehicle/proc/absorbDamage(damage,damage_type)
+	var/coeff = 1
+	var/obj/item/vehicle_parts/armor/A = locate() in parts
+	if(A.vehicle_armor[damage_type])
+		coeff = A.vehicle_armor[damage_type]
+	return damage*coeff
+
+/obj/vehicle/examine(mob/user)
+	..()
+	var/integrity = health/maxhealth*100
+	switch(integrity)
+		if(85 to 100)
+			user << "It's fully intact."
+		if(65 to 85)
+			user << "It's slightly damaged."
+		if(45 to 65)
+			user << "It's badly damaged."
+		if(25 to 45)
+			user << "It's heavily damaged."
+		else
+			user << "It's falling apart."
+	if(parts)
+		user << "It's has:"
+		for(var/obj/item/part in parts)
+			if(istype(part,/obj/item/weapon/reagent_containers/fueltank))
+				var/obj/item/weapon/reagent_containers/fueltank/check = part
+				var/list/rbeaker = list()
+				for(var/datum/reagent/R in check.reagents.reagent_list)
+					rbeaker += R.name
+				var/contained = english_list(rbeaker)
+				user << "\icon[part] [part] and has [check.reagents.total_volume] out of a max [check.volume], containing [contained]"
+			else
+				user << "\icon[part] [part]"
+
+
+/obj/vehicle/proc/GrantActions(var/mob/living/user)
+	exit_vehicle.vehicle = src
+	exit_vehicle.Grant(user)
+
+/obj/vehicle/proc/RemoveActions(var/mob/living/user)
+	exit_vehicle.Remove(usr)
+
+
+
+/obj/vehicle/proc/exit(var/atom/newloc = loc)
+	RemoveActions()
+	usr.forceMove(newloc)
+	if(usr == driver)
+		driver = null
+	else
+		occupants--
+	usr.reset_view()
+
+/obj/vehicle/Destroy(var/atom/newloc = loc)
+	for(var/mob/M in src)
+		RemoveActions(M)
+		M.forceMove(newloc)
+		M.reset_view()
+	..()
+
+/datum/action/vehicle
+	check_flags = AB_CHECK_RESTRAINED | AB_CHECK_STUNNED | AB_CHECK_ALIVE
+	action_type = AB_INNATE
+	var/obj/vehicle/vehicle
+
+/datum/action/vehicle/exit_vehicle
+	name = "Leave Vehicle"
+	button_icon_state = "mech_eject"
+
+/datum/action/vehicle/exit_vehicle/Activate()
+	if(!owner || !iscarbon(owner))
+		return
+	if(!vehicle)
+		return
+	vehicle.exit()
+
+/obj/vehicle/New()
+	for(var/obj/item/O in parts)
+		O.loc = src
+	var/obj/item/vehicle_parts/engine/E = locate() in parts
+	var/obj/item/weapon/reagent_containers/fueltank/F = locate() in parts
+	F.reagents.add_reagent(E.fueltype, F.volume) //Debug stuff
```

Comment:
Thats only there to fill the fuel tank up to max when i spawn the stupid thing, it will be taken out before this is merged.